pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -502,13 +503,11 @@ func (c *Controller) addOrUpdateService(pre, curr *v1.Service, currConv *model.S
 		}
 	}
 
-	// filter out same service event
-	if event == model.EventUpdate && !serviceUpdateNeedsPush(pre, curr, prevConv, currConv) {
-		return
-	}
-
 	c.opts.XDSUpdater.SvcUpdate(shard, string(currConv.Hostname), ns, event)
```

Comment:
Hm, interesting that this essentially does nothing. SvcUpdate only handles `Delete`, and this code is only on Update/Create

Still probably a good change though, we do increment a metric, and in the future its plausible SvcUpdate will do something outside of delete I suppose